securesign/secure-sign-operator
# feat: Add service monitors to rekor and fulcio

This pr aims to add the service monitors for rekor and fulcio.

# Comments
In order for me to create the service monitor I needed to import `github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1`. This pkg needs an updated version of sigs.k8s.io/controller-runtime to run and in this version MetricsBindAddress and Port where removed https://github.com/securesign/secure-sign-operator/blob/59a3a27241d3157c141fc5846e0741824f1b1884/main.go#L80 Ive commented them out for now but let me know if this is ok?

Source: https://github.com/operator-framework/operator-sdk/issues/6608

## Testing
1. Do usual Setup Cluster, Login, Keycloak, update yaml file with Keycloak URL.
2. Create a ns with the cluster monitoring label: `oc create ns openshift-trusted-artifact-signer && oc label namespace openshift-trusted-artifact-signer openshift.io/cluster-monitoring=true`
3. Apply yaml file with `enableMonitoring: true`.
4. Wait for a bit then go to Observe --> Targets, eventual you should see two targets, `fulcio-metrics` & `rekor-metrics`
5. These metrics can be queried by going to Observe --> Metrics
